### ISSUE TITLE
Be more precise than generic `Callable` in type hints

### ISSUE THREAD

**Issue by benoit74** [openzim/python-scraperlib#199]
`callback` argument of `zim.creater.Creator.add_item_for` is reported by pyright as partially unknown when in `strict` mode due to the use of generic `Callable`.

This is a problem in projects setting `pyright` in strict mode.

The same generic `Callable` seems to be an issue in other parts of the code.

We should probably fix this with a more precise type hint,

**Comment by rgaudin** [comment]
Indeed, we should define a signature/interface for it. It's not like it's used in many ways anyway

**Comment by benoit74** [comment]
Callable typing issue are fixed with #221, #227 and soon #220 . We might want to define a signature/interface, but this is beyond 5.0.0 for me, so postponing to later.
